watcher/watcher.go

```diff
@@ -288,20 +288,27 @@ func (w *Watcher) reportRoomsStatusesRoutine() {
 }
 
 func (w *Watcher) watchRooms() error {
-	l := w.Logger.WithFields(logrus.Fields{
-		"operation": "watcher.watchRooms.EnsureCorrectRooms",
-	})
-	_, _ = w.WithDownscalingLock(l, w.EnsureCorrectRooms)
+	_, _ = w.WithDownscalingLock(
+		w.Logger.WithFields(logrus.Fields{
+			"operation": "watcher.watchRooms.EnsureCorrectRooms",
+		}),
+		w.EnsureCorrectRooms,
+	)
 
-	l = w.Logger.WithFields(logrus.Fields{
-		"operation": "watcher.watchRooms.RemoveDeadRooms",
-	})
-	_, _ = w.WithDownscalingLock(l, w.RemoveDeadRooms)
+	go w.WithRemoveDeadRoomsLock(
```

Comment:
WDYT of moving this goroutine to run as first instruction? Above of EnsureCorrectRooms.. makes sense?

Reply:
There will be differences only on the first execution. But we can change it.